client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -299,93 +340,102 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
-                          index: index + 1,
-                        },
-                        ''
-                      )
-                    : index + 1}
-                </Typography>
-              </Box>
-            </AccordionSummary>
-            <AccordionDetails>
-              <JsonFormsDispatch
-                key={childPath}
-                schema={schema}
-                uischema={childUiSchema || uischema}
-                enabled={enabled}
-                path={childPath}
-                renderers={renderers}
-              />
-            </AccordionDetails>
-          </Accordion>
-        );
-      })}
+                  <ConfirmationModal
+                    open={removeIndex !== undefined}
+                    onConfirm={() => {
+                      if (removeIndex !== undefined) {
+                        removeItems(path, [removeIndex])();
+                        setRemoveIndex(undefined);
+                      }
+                    }}
+                    onCancel={() => setRemoveIndex(undefined)}
+                    title={t('label.remove')}
+                    message={t('messages.confirm-remove-item')}
+                  />
+
+                  <IconButton
+                    icon={isEditable ? <MinusCircleIcon /> : null}
+                    label={t('label.remove')}
+                    color="primary"
+                    className="array-remove-icon"
+                    sx={{ visibility: 'hidden' }}
+                    onClick={e => {
+                      setRemoveIndex(index);
+                      // Don't toggle the accordion:
+                      e.stopPropagation();
+                    }}
+                  />
+
+                  <Typography
+                    sx={{
+                      fontWeight: 'bold',
+                      textAlign: 'end',
+                      whiteSpace: 'nowrap',
+                    }}
+                  >
+                    {uischema?.itemLabel
+                      ? RegexUtils.formatTemplateString(
+                          uischema?.itemLabel,
+                          {
+                            ...(typeof child === 'object' ? child : {}),
+                            index: index + 1,
+                          },
+                          ''
+                        )
+                      : index + 1}
+                  </Typography>
+                </Box>
+              </AccordionSummary>
+              <AccordionDetails>
+                <JsonFormsDispatch
+                  key={childPath}
+                  schema={schema}
+                  uischema={childUiSchema || uischema}
+                  enabled={isEditable}
+                  path={childPath}
+                  renderers={renderers}
+                />
+              </AccordionDetails>
+            </Accordion>
+          );
+        })
+        .sort((a, b) =>
+          uischema.options?.['reverse']
```

Comment:
As above we should at least have a type definition for the options and cast the `any` type to this type definition, preferably with zod validation.

Maybe replace `data` and use Array.revert()?  I also thought about reverting the input `data` array before creating the array of Components but I am not sure if this would have some side effects...

Reply:
Have added a complete Zod schema for options and validation check in now.